### PR TITLE
Improve long term caching ability

### LINKED ISSUE #47: Webpack `compiler_vendor`
Our boilerplate's config has a section for `compiler_vendor` that is currently unused in our new webpack config.
We should investigate the features of this option and look for potential optimizations or drop the section from the config.

### PR DESCRIPTION
**Pull request by thomaseizinger**
Basically, this PR applies the caching guide of webpack (https://webpack.js.org/guides/caching/) to our project.

Closes #47.

### CODE REVIEW

File: build/webpack.config.js [hunk 1]
```diff
@@ -121,7 +121,7 @@ let webpackConfig = {
         test: /\.(png|jpg)$/,
         loader: 'url-loader',
         options: {
-          name: `[name].[hash].[ext]`,
+          name: `[name].[hash:4].[ext]`,
```
Comment (duffleit) [inline build/webpack.config.js]: http://bit.ly/2eZq9Ni

Comment (thomaseizinger) [reply build/webpack.config.js]: Hehe, I just thought that the filenames are a bit long and I therefore shortened the hash to 4 chars.

**Comment by thomaseizinger** [comment]
Did you read the guide I linked to?
If you want, we can go through the config at some point.

**Comment by thomaseizinger** [comment]
The basic idea is the following:
webpack allows to define several entry points. To my understanding, each entry point becomes a bundle. With the help of the CommonsChunkPlugin, you can define which bundles go into which chunks. A chunk is thereby a .js file that is outputted by webpack.
If we create a dedicated entry point for all the vendor packages (i.e. long-term dependencies), we can achieve that the generated bundle (and its hash) does not change over several releases. Hence, users don't have to download the whole app again but just the files that have changed (only our application code and application styles).

**Pull request merged**